### PR TITLE
Test mongo adaptor on CI

### PR DESCRIPTION
**Pull request by anthonyalberto**
A bunch of stuff happening here.

TL;DR:

- We now have a proper way of running adaptor tests locally or in CI
- Mongo adaptor tests fixed, they now run in CI
- Regenerated a bunch of TLS certs to make mongo TLS tests pass, documented it

# Harmonize how we run adaptor tests

This provides a unique way to run adaptor tests, for either local dev or CI `scripts/run_db_in_docker.sh <adaptor> <version>` will be in charge of spinning up the required containers via docker-compose.

This PR provides everything necessary to run mongodb with this script.

Once the script is done, you can just `go test -v ./adaptor/mongodb/`

This PR runs mongodb adaptor tests on Github Actions, and they pass!

# Fix mongodb tests, TLS shenanigans

MongoDB tests were totally broken. Spent a long time with TLS tests not working. As it turns out, golang now requires certs with AltText tags otherwise it just fails. I therefore regenerated certs and documented how I did it.

Another, bigger problem was, connecting to mongo from within the docker container and your local machine via IP wouldn't work because of some mongo TLS restrictions. Basically, when you boot mongodb, you have to pass an IP or hostname to bind to. If you then try to connect to that mongo over TLS with a cert that does not match that IP or hostname, it fails.
I had to use the local network IP in order to expose ports via docker, but then use `127.0.0.1` to connect to them from my local machine. 

So I introduced a hostname, `transporter-db` and bind to that. From within the container, it points to the local network IP. From within your local machine, it will point to `127.0.0.1`. Had to update the tests to use `transporter-db` too.

With all that said, a single test couldn't pass still. I'm 99% confident it's an issue with `mgo`, the driver we're using (it's not maintained anymore). I've written 3 tests scripts (bash, ruby, go) to connect with TLS with verification with a self signed cert, and only the golang one failed even though all of them used the same certs.

We'll need to try and replace the mongo driver to the official one at some point I guess.

Finally, mentioning that in the past we were doing objects comparisons in tests with `reflect.DeepEqual` on `tls.config` objects and it appeared to work fine. Something must have changed with modern go, as `tls.config` aren't comparable like that anymore, so had to write some logic and a function to work around that.

### CODE REVIEW

**Comment by pete-may** [comment]
Hey when running tests locally I'm getting:
```
--- FAIL: TestTail (12.66s)
```
With errors like `reader_test.go:334: [insertDeleteData] bad message count, expected 2, got 0`

And also:
```
time="2021-11-08T14:34:11-05:00" level=info msg="testing oplog access" uri="mongodb://127.0.0.1:27017/test"
    client_test.go:509: [with_tail] unexpected Connect error, expected <nil>, got oplog access failed, database missing oplog.rs collection
--- FAIL: TestConnect (3.56s)
```
which I think may be the culprit. So something wrong with reading the test db's oplog?

**Comment by pete-may** [comment]
I think it has to do with this line using `DefaultURI`. 
https://github.com/compose/transporter/blob/852e9006eb39c8af4f30b12d4d2240f74f040ba8/adaptor/mongodb/client_test.go#L462

The `DefaultURI` is defined here: https://github.com/compose/transporter/blob/9e154e76b7d2977d9ac7756660779b512cace87f/adaptor/mongodb/client.go#L23

And I have a local mongo running there already.

**Comment by pete-may** [comment]
Yup I switched off my local mongo that was taking up the `27017` port on my localhosts and then tests passed. Not sure if we should use a different port or try to warn that there is already a mongo instance running when tests are run.

**Comment by anthonyalberto** [comment]
I'll mention the list of ports needed for docker-compose on the mongo adaptor readme

**Review by pete-may — APPROVED** [review]
Cool! LGTM
